open-access-variant-filtering-wf/main.nf

```diff
@@ -0,0 +1,199 @@
+params.score_url = ""
+params.cleanup = true
+
+params.download = [:]
+params.upload = [:]
+params.filter = [:]
+params.payloadGenVcf = [:]
+
+download_params = [
+    'song_cpus': params.cpus,
+    'song_mem': params.mem,
+    'score_cpus': params.cpus,
+    'score_mem': params.mem,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    'publish_dir': params.publish_dir,
+    *:(params.download ?: [:])
+]
+
+upload_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    'publish_dir': params.publish_dir,
+    *:(params.upload ?: [:])
+]
+
+filter_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'publish_dir': params.publish_dir,
+    'regions_file': params.regions_file,
+    'apply_filters': params.apply_filters,
+    'output_type': params.output_type,
+    'include': params.include,
+    'exclude': params.exclude,
+    *:(params.filter ?: [:])
+]
+
+payloadGenVcf_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'publish_dir': params.publish_dir,
+    *:(params.payloadGenVcf ?: [:])
+]
+
+include { songScoreDownload as dnVcf } from './song-score-utils/song-score-download' params(download_params)
+include { metadataParser as mParser } from "./modules/raw.githubusercontent.com/icgc-argo/data-processing-utility-tools/metadata-parser.0.2.0.0/tools/metadata-parser/metadata-parser.nf"
+include { variantFilter as vFilter } from './wfpr_modules/github.com/icgc-argo/variant-calling-tools/variant-filter@0.1.0/main.nf' params(filter_params)
+include { payloadGenVariantProcessing as pGenVar } from "./wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/payload-gen-variant-processing@0.1.0/main.nf" params(payloadGenVcf_params)
+include { songScoreUpload as upVcf } from './song-score-utils/song-score-upload' params(upload_params)
+include { getSecondaryFiles as getSec } from './wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/helper-functions@1.0.0/main.nf'
+include { cleanupWorkdir as cleanup } from './wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/cleanup-workdir@1.0.0/main.nf'
+
+
+// please update workflow code as needed
+workflow OpenFilterWf {
+  take:  // update as needed
+    study_id
+    analysis_id
+    vcf_file
+    analysis_metadata
+    regions_file
+
+
+  main:  // update as needed
+    local_mode = false
+    vcf = Channel.from()
+    vcf_idx = Channel.from()
+
+    if (analysis_id) {
+      // download vcf and metadata from song/score (analysis type: variant_calling)
+      dnVcf(study_id, analysis_id)
+      vcf = dnVcf.out.files.flatten().first()
+      vcf_idx = dnVcf.out.files.flatten().last()
+      vcf_meta = dnVcf.out.song_analysis
+    } else if (
+      !analysis_metadata.startsWith('NO_FILE') && \
+      !vcf_file.startsWith('NO_FILE')
+    ) {
+      local_mode = true
```

Comment:
here, we can add a check to make sure `publish_dir` is set. Running local mode, it does not make much sense to not set it.